giraffe/src/components/SimpleTable/PagedTable.tsx

```diff
@@ -40,24 +38,24 @@ const getNumberOfRowsOnCurrentPage = (
   headerHeight: number,
   rowHeight: number
 ): number => {
-  if (totalAvailableHeight === 0) {
+  if (totalAvailableHeight <= 0) {
     return 0
   }
 
-  const minimumLength = result?.table?.length ?? 1
-  const estimatedRowHeight = Math.min(
-    Math.ceil(totalAvailableHeight / minimumLength),
-    MAXIMUM_ESTIMATED_ROW_HEIGHT
-  )
-  const visibleRowHeight = Math.max(rowHeight, estimatedRowHeight)
+  // this means that no rows have been mounted or measured, so we need to
+  // mount one row to measure the row height
+  if (rowHeight <= 0) {
+    return 1
```

Comment:
this seems like a smell. It seems like an indication that this function is doing too much, or isn't doing what it says its doing when we have to return 1 when there is no row height. Seems like we have more refactoring to do, because to me, this doesn't make a lot of logical sense

Reply:
There are further refactor changes pending.

This pull request addresses the issue of the `usePlotEnv` hook dramatically slowing down Simple Table which was found by @asalem1